heroku-18-cnb-build/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM heroku/heroku:18-build
+
+ARG pack_uid=1000
+ARG pack_gid=1000
+
+RUN groupadd heroku --gid ${pack_gid} && \
+  useradd heroku -u ${pack_uid} -g ${pack_gid} -s /bin/bash -m
+RUN mkdir /app && \
+  chown heroku:heroku /app
+
+ENV CNB_USER_ID=${pack_uid}
+ENV CNB_GROUP_ID=${pack_gid}
+
+ARG STACK="heroku-18"
+ENV STACK "${STACK}"
```

Comment:
From some git rummaging, it seems `STACK` was first added to the builder image in 2018, in:
https://github.com/heroku/pack-images/commit/13a9f38416ebe85156e52bfd3b7f2aa4c0da3e86

However cnb-shim didn't start setting `STACK` until 2020, in:
https://github.com/heroku/cnb-shim/pull/3

The description of that PR says it's to support heroku/pack-images#78, which led me to:
https://github.com/buildpacks/lifecycle/releases/tag/v0.7.0

And thus:

> The platform must provide environment variables to buildpacks using /platform/env, rather than setting them directly in the lifecycle's environment. This brings it in compliance with [the specification](https://github.com/buildpacks/spec/blob/master/buildpack.md#provided-by-the-platform).

From the current spec:
https://github.com/buildpacks/spec/blob/main/platform.md#buildpack-environment

> The platform SHOULD NOT assume any other stack-provided environment variables are inherited by the buildpack.

As such, it seems like removing `STACK` from the build image will actually not have an effect on buildpacks regardless, since `lifecycle` only allow-lists certain env vars to be propagated.